corehq/apps/reports/management/commands/create_tableau_connection.py

```diff
@@ -0,0 +1,220 @@
+                 """,
+        )
+        parser.add_argument(
+            '--PAT-secret',
+            action='store',
+            dest='pat_secret',
+            required=True,
+            help="The Personal Access Token secret.",
+        )
+        parser.add_argument(
+            '--server-name',
+            action='store',
+            dest='server_name',
+            required=True,
+            help="The host name of the server that is running Tableau.",
+        )
+        parser.add_argument(
+            '--site-name',
+            action='store',
+            dest='site_name',
+            required=True,
+            help="The Tableau site to create a connected app on.",
+        )
+        parser.add_argument(
+            '--domain',
+            action='store',
+            dest='domain',
+            required=True,
+            help="The name of the domain to use connect the Tableau site to, for use with Embedded Tableau.",
+        )
+
+    def handle(self, **options):
+        self.options = options
+        self.server_name = self.options['server_name']
+        self.site_name = self.options['site_name']
+        self.domain = self.options['domain']
+
+        api_version = 3.14  # Minimum version needed to use the method to create a connected app.
+        self.url_base = f"https://{self.server_name}/api/{api_version}"
+
+        # Without these headers, the Tableau API uses XML by default.
+        self.headers = {
+            "Content-Type": "application/json",
+            "Accept": "application/json"
+        }
+
+        # You must be signed into an authenticated account to create a Connected App with the Tableau API.
+        self.sign_in()
+        try:
+            # The site ID is necessary for making site-specific API requests.
+            self.site_id = self.get_site_id()
+            (connected_app_exists, connected_app_client_id) = self.check_if_connected_app_exists()
+            if connected_app_exists:
+                overwrite = input("\nA CommCareHQ connected app already exists on this site. Overwrite and "
+                                  "create a new one? [y/n]")
+                if overwrite == 'y':
+                    self.delete_existing_connected_app(connected_app_client_id)
+                else:
+                    logger.info("No app created. Command execution completed.")
+                    self.sign_out()
+                    return
+            self.create_connected_app()
+            logger.info("App succesffully created. Command execution completed.")
+            self.sign_out()
+        # I think it's ok to ignore the flake8 error for this bare except clause since we re-raise the exception.
+        except:  # noqa: E722
+            logger.info("Command execution interrupted.")
+            # Submit a sign out request on any interruption to ensure the token we were given upon sign in is no
+            # longer valid.
+            self.sign_out()
+            raise
```

Comment:
While I think this is fine as-is, this pattern would be a good place to try a `with` statement if you feeling like playing with context managers.